endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java

```diff
@@ -120,6 +121,24 @@ public boolean hasMessageListeners() {
      * @return
      */
     private String getRequestContent(HttpRequest request, String body) {
+        String contentType = request.getHeaders().getFirst(HttpHeaders.CONTENT_TYPE);
+        if (contentType != null) {
+            String[] contentTypeParts = contentType.split(";");
+            for (String contentTypePart : contentTypeParts) {
+                if(contentTypePart.startsWith("charset=")) {
+                    String charset = contentTypePart.split("=")[1];
```

Comment:
if for some reason the header contains only `charset=`? this line will fail. please add a check `contentTypePart.split("=").length == 2`.

Reply:
prevent failure by adding `!contentTypePart.endsWith("charset=")` to if clause.